smithy-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/lang/DocumentationPreprocessorTest.kt

```diff
@@ -68,4 +68,493 @@ class DocumentationPreprocessorTest {
             assertEquals(expected, docs)
         }
     }
+
+    @Test
+    fun `it renders paragraphs`() {
+        val input = "<p>this is paragraph</p><div>this too is paragraph</div>"
+        val expected = """
+        this is paragraph
+        
+        this too is paragraph
+        
```

Comment:
There can be a few at the end because the transformer doesn't check whether the document is ending when it terminates a paragraph, list, etc.

The existing `KotlinWriter` implementation will drop trailing whitespace like this.

Reply:
I would expect the preprocessor to cleanly return markdown that doesn't need to be processed again (i.e., by trimming whitespace).